substrate/frame/dap/src/lib.rs

```diff
@@ -0,0 +1,389 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Dynamic Allocation Pool (DAP) Pallet
+//!
+//! This pallet implements `FundingSink` to collect funds into a buffer account instead of burning
+//! them. The buffer account is created via `inc_providers` at genesis, ensuring it can receive any
+//! amount including those below ED.
+//!
+//! For existing chains adding DAP, include `dap::migrations::InitBufferAccount` as a permanent
```

Comment:
If we explicitly increase the provider, unless we decrease it, it will always exist. And since its just a pallet account (not spammable), thats fine to leave a provider on it. We don't have to keep it as permanent.